internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -206,26 +207,63 @@ func (r *BundleReconciler) ensureUnpackPod(ctx context.Context, bundle *rukpakv1
 		pod.Spec.AutomountServiceAccountToken = &automountServiceAccountToken
 		pod.Spec.Volumes = []corev1.Volume{
 			{Name: "util", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
+			{Name: "manifests", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
 		}
 		pod.Spec.RestartPolicy = corev1.RestartPolicyNever
-		if len(pod.Spec.InitContainers) != 1 {
-			pod.Spec.InitContainers = make([]corev1.Container, 1)
+		if bundle.Spec.Source.Type == "git" {
+			// Note: use host networking (hack) so we can resolve https://github.com references in the
+			// Bundle's spec.Source.Git.Repository. Maybe we should revisit having the user inline https://github.com?
+			pod.Spec.HostNetwork = true
+		}
+
+		targetInitContainers := 1
+		if bundle.Spec.Source.Type == "git" {
+			targetInitContainers = 2
 		}
 		pod.Spec.InitContainers[0].Name = "install-unpacker"
+		if len(pod.Spec.InitContainers) != targetInitContainers {
+			pod.Spec.InitContainers = make([]corev1.Container, targetInitContainers)
+		}
 		pod.Spec.InitContainers[0].Image = r.UnpackImage
 		pod.Spec.InitContainers[0].ImagePullPolicy = corev1.PullIfNotPresent
 		pod.Spec.InitContainers[0].Command = []string{"cp", "-Rv", "/unpack", "/util/unpack"}
 		pod.Spec.InitContainers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}}
 
+		// Note: initContainer so we can ensure the repository has been cloned
+		// at the bundle.Spec.Source.Git.Ref before we unpack the Bundle contents
+		// that are stored in the repository.
+		if bundle.Spec.Source.Type == "git" {
+			pod.Spec.InitContainers[1].Name = "clone-repository"
+			pod.Spec.InitContainers[1].Image = "bitnami/git:latest"
+			pod.Spec.InitContainers[1].ImagePullPolicy = corev1.PullIfNotPresent
+			checkoutCmd := git.NewCheckoutCmd(*bundle.Spec.Source.Git)
+			c, err := checkoutCmd.String()
+			if err != nil {
+				return err
+			}
+			pod.Spec.InitContainers[1].Command = []string{"/bin/bash", "-c", c}
+			pod.Spec.InitContainers[1].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}, {Name: "manifests", MountPath: "/manifests"}}
+		}
+
 		if len(pod.Spec.Containers) != 1 {
 			pod.Spec.Containers = make([]corev1.Container, 1)
 		}
+
 		if bundle.Spec.Source.Image != nil {
 			pod.Spec.Containers[0].Name = bundleUnpackContainerName
 			pod.Spec.Containers[0].Image = bundle.Spec.Source.Image.Ref
 			pod.Spec.Containers[0].Command = []string{"/bin/unpack", "--bundle-dir", "/"}
 			pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/bin"}}
 		}
+
+		if bundle.Spec.Source.Type == "git" {
+			pod.Spec.Containers[0].Name = bundleUnpackContainerName
+			pod.Spec.Containers[0].Image = "bitnami/git:latest"
```

Comment:
follow-up: make this image configurable?

Reply:
Any specific reason we chose this one? I also see https://hub.docker.com/r/alpine/git, which looks quite a bit smaller and it supports multi-arch.

Looks like I wouldn't be able to run bitnami/git on my M1 mac without emulation since it is only a linux/amd64 machine.

Also, should we pin to a version (or version stream)?

Reply:
I think we should also consider a pure in-process Go implementation to avoid having to worry about this extra image. But I think we can defer that until/after we get a `Source` interface in place.

Reply:
We can use the alpine git based image sure -- and I agree it should be configurable. 

Doing the git unpacking in process is also doable once we have the abstractions in place.